src/fairseq2/recipes/wav2vec2/train.py

```diff
@@ -0,0 +1,360 @@
+
+    checkpoint_after_n_steps: int = 0
+    """The number of steps after which to start checkpointing."""
+
+    checkpoint_every_n_steps: int = 25_000
+    """The step interval at which to checkpoint."""
+
+    publish_metrics_every_n_steps: int = 1000
+    """The step interval at which to publish metrics."""
+
+    # Misc
+    seed: int = 2
+    """The random number generator seed to use."""
+
+    profile: Optional[Tuple[int, int]] = None
+    """The number of steps that the PyTorch profiler should skip and then record."""
+
+    monitored_gang: bool = False
+    """If ``True``, puts a monitored barrier before every collective call."""
+
+    anomaly_detection: bool = False
+    """If ``True``, enables the anomaly detection feature of ``torch.autograd``."""
+
+
+wav2vec2_train_presets = ConfigRegistry[Wav2Vec2TrainConfig]()
+
+wav2vec2_train_preset = wav2vec2_train_presets.decorator
+
+
+@wav2vec2_train_preset("base_960h")
+def _base_960h() -> Wav2Vec2TrainConfig:
+    return Wav2Vec2TrainConfig()
+
+
+def load_wav2vec2_trainer(
+    config: Wav2Vec2TrainConfig, output_dir: Path
+) -> Trainer[Tensor]:
+    """Load a :class:`Trainer` for wav2vec 2.0 model training."""
+    wall_watch = Stopwatch(start=True)
+
+    gang = setup_root_gang(log, monitored=config.monitored_gang)
+
+    checkpoint_manager = FileCheckpointManager(output_dir.joinpath("checkpoints"), gang)
+
+    # Load the dataset.
+    dataset_card = retrieve_asset_card(config.dataset)
+
+    log.info("Loading {} speech dataset.", dataset_card.name)
+
+    dataset = load_speech_dataset(config.dataset)
+
+    log.info("Dataset loaded.")
+
+    # Initialize the model configuration.
+    if config.model_arch is None:
+        model_config = Wav2Vec2Config()
+    else:
+        model_config = wav2vec2_archs.get(config.model_arch)
+
+    if config.model_config is not None:
+        update_model_config(model_config, config.model_config)
+
+    log_model_config(model_config, log)
+
+    # Initialize the model.
+    model = create_wav2vec2_model(model_config, device=META, dtype=torch.float32)
+
+    checkpoint_manager.save_model_metadata(family=model.family, config=model_config)
+
+    has_checkpoint = checkpoint_manager.has_checkpoint()
+
+    # If we don't have a checkpoint, load the model on rank 0 and
+    # broadcast it to the gang.
+    if not has_checkpoint:
+        if gang.rank == 0:
+            to_device(model, gang.device, seed=config.seed)
+
+        gang.barrier()
+
+    dp_model = to_data_parallel(
+        model,
+        gang,
+        config.data_parallelism,
+        log,
+        ddp_find_unused_parameters=False,
+        fsdp_skip_init=True,
+        fsdp_broadcast_state=not has_checkpoint,
+        fsdp_mixed_precision_dtype=config.dtype,
+        fsdp_fp32_reduce=True,
+        fsdp_wrap_granularity=config.fsdp_wrap_granularity,
+    )
+
+    if config.torch_compile:
+        model.encoder = compile_model(model.encoder, log)  # type: ignore[assignment]
```

Comment:
should this be `ddp_model.encoder` ?

Reply:
@cbalioglu we should fix this in the ASR pipeline too right?